packages/EasyCore/tests/Bridge/Symfony/Serializer/TrimStringsNormalizerTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyCore\Tests\Bridge\Symfony\Serializer;
+
+use EonX\EasyCore\Bridge\Symfony\Serializer\TrimStringsDenormalizer;
+use EonX\EasyCore\Tests\Bridge\Symfony\AbstractSymfonyTestCase;
+use EonX\EasyCore\Tests\Bridge\Symfony\Stubs\DenormalizerInterfaceStub;
+use EonX\EasyCore\Tests\Helpers\CleanerInterface;
+use stdClass;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+final class TrimStringsNormalizerTest extends AbstractSymfonyTestCase
+{
+    /**
+     * @return mixed[]
+     *
+     * @see testSupportsDenormalizationSucceeds
+     */
+    public function provideDataForSupportsDenormalization(): array
+    {
+        return [
+            'data is string' => [
+                '',
+                true,
+            ],
+            'data is array' => [
+                [],
+                true,
+            ],
+            'data is bool' => [
+                true,
+                false,
+            ],
+            'data is numeric' => [
+                123,
+                false,
+            ],
+            'data is object' => [
+                new stdClass(),
+                false,
+            ],
+        ];
+    }
+
+    /**
+     * @param mixed $data
+     *
+     * @dataProvider provideDataForSupportsDenormalization
+     */
+    public function testSupportsDenormalizationSucceeds($data, bool $expectedResult): void
+    {
+        $cleaner = $this->prophesize(CleanerInterface::class);
```

Comment:
Why do you use Mockery in one test class and Prophecy in another one? Let's keep it consistent and use Mockery everywhere.

Reply:
Done.

Reply:
I think we could use a real implementation here (TrimStringsRecursive). Also we can use DenormalizerInterfaceStub.

Reply:
Done.